crates/orderbook/src/api.rs

```diff
@@ -32,48 +33,110 @@ pub fn handle_all_routes(
     orderbook: Arc<Orderbook>,
     quoter: Arc<OrderQuoter>,
 ) -> impl Filter<Extract = (impl Reply,), Error = Rejection> + Clone {
-    let create_order = create_order::create_order(orderbook.clone()).boxed();
-    let get_orders = get_orders::get_orders(orderbook.clone()).boxed();
-    let fee_info = get_fee_info::get_fee_info(quoter.fee_calculator.clone()).boxed();
-    let get_order = get_order_by_uid::get_order_by_uid(orderbook.clone()).boxed();
-    let get_solvable_orders = get_solvable_orders::get_solvable_orders(orderbook.clone()).boxed();
-    let get_solvable_orders_v2 =
-        get_solvable_orders_v2::get_solvable_orders(orderbook.clone()).boxed();
-    let get_trades = get_trades::get_trades(database).boxed();
-    let cancel_order = cancel_order::cancel_order(orderbook.clone()).boxed();
-    let get_amount_estimate =
-        get_markets::get_amount_estimate(quoter.price_estimator.clone()).boxed();
-    let get_fee_and_quote_sell = get_fee_and_quote::get_fee_and_quote_sell(quoter.clone()).boxed();
-    let get_fee_and_quote_buy = get_fee_and_quote::get_fee_and_quote_buy(quoter.clone()).boxed();
-    let get_user_orders = get_user_orders::get_user_orders(orderbook.clone()).boxed();
-    let get_orders_by_tx = get_orders_by_tx::get_orders_by_tx(orderbook).boxed();
-    let post_quote = post_quote::post_quote(quoter).boxed();
+    // api/v1
+    let create_order = (
+        "/api/v1/orders",
+        create_order::create_order(orderbook.clone()),
```

Comment:
I'm not sure if it's better to store the `"/api/v1/orders"` string here, or if `create_order::create_order` should return it as well. The concern here is that `create_order::create_order` doesn't know about the `"/api/v1"` prefix, hence it shouldn't return the `"/api/v1/orders"` string. On the other hand, we shouldn't rely on the contents of `create_order::create_order` function on the call site, hence we can't rely on the fact that it returns a filter for path `"/orders"`.

Reply:
One solution could be to move the `/api/v1/` part of the url into the `create_order::create_order` function.